src/gazebo_mavlink_interface.cpp

```diff
@@ -957,6 +962,65 @@ void GazeboMavlinkInterface::IRLockCallback(IRLockPtr& irlock_message) {
   mavlink_interface_->send_mavlink_message(&msg);
 }
 
+void GazeboMavlinkInterface::ArucoMarkerCallback(ArucoMarkerPtr& arucoMarker_message) {
+  
+  mavlink_target_relative_t sensor_msg;
+  sensor_msg.timestamp = arucoMarker_message->time_usec();
+  sensor_msg.x = arucoMarker_message->pos_x();
+  sensor_msg.y = arucoMarker_message->pos_y();
+  sensor_msg.z = arucoMarker_message->pos_z();
+
+  sensor_msg.type = LANDING_TARGET_TYPE_VISION_FIDUCIAL;
+  sensor_msg.frame = TARGET_OBS_FRAME_BODY_FRD;
+
+  sensor_msg.q_sensor[0] = arucoMarker_message->attitude_q_w();
+  sensor_msg.q_sensor[1] = arucoMarker_message->attitude_q_x();
+  sensor_msg.q_sensor[2] = arucoMarker_message->attitude_q_y();
+  sensor_msg.q_sensor[3] = arucoMarker_message->attitude_q_z();
+
+  sensor_msg.pos_var[0] = arucoMarker_message->var_x();
+  sensor_msg.pos_var[1] = arucoMarker_message->var_y();
+  sensor_msg.pos_var[2] = arucoMarker_message->var_z();
+
+  sensor_msg.q_target[0] = arucoMarker_message->orientation_q_w();
+  sensor_msg.q_target[1] = arucoMarker_message->orientation_q_x();
+  sensor_msg.q_target[2] = arucoMarker_message->orientation_q_y();
+  sensor_msg.q_target[3] = arucoMarker_message->orientation_q_z();
+
+  sensor_msg.yaw_var = arucoMarker_message->yaw_var();
+
+  mavlink_message_t msg;
+  mavlink_msg_target_relative_encode_chan(1, 200, MAVLINK_COMM_0, &msg, &sensor_msg);
+  mavlink_interface_->send_mavlink_message(&msg);
+}
+
+void GazeboMavlinkInterface::TargetGpsCallback(GpsPtr& gps_msg) {
+
+  mavlink_target_absolute_t gps_data;
```

Comment:
These mavlink messages are not merged (https://github.com/mavlink/mavlink/pull/1943) Therefore, I don't think we can merge this PR into this repository

Reply:
Give it a few more days :smile:

Reply:
@potaito Sure :grin: